### PR TITLE
Fix: 1208. Intermediaries altering content/transfer-coding or content…

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- [x] avoid ambiguous statement on digest protecting from compressed content-coding https://github.com/httpwg/http-extensions/issues/1208#issuecomment-634351643
- i[x] t can protect from intermediaries altering representation metadata (content-type, content-coding)

### CODE REVIEW

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ integrity protection; for instance TCP checksums or TLS records [RFC2818].
 
 However, there are cases where relying on this alone is insufficient. An
 HTTP-level integrity mechanism that operates independent of transfer can be used
-to detect programming errors and/or corruption of data at rest, be used across
+to detect programming errors and/or corruption of data in flight or at rest, be used across
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:120]: adding "in flight" seems werid. What errors do you expect that TLS integrity wouldn't have caught?

File: draft-ietf-httpbis-digest-headers.md [hunk 2]
```diff
@@ -281,6 +281,7 @@ A recipient MAY ignore any or all of the representation-data-digests in a Digest
 field. This allows the recipient to choose which digest-algorithm(s) to use for
 validation instead of verifying every received representation-data-digest.
 
+An intermediary MUST NOT alter the value of a received `Digest` field.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:284]: Do we need this? Isn't it just restating a requirement about end-to-end headers? 

Perhaps we can jsut reference to semantics here too.

File: draft-ietf-httpbis-digest-headers.md (hunk 1 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:120]: It refers to trasfer-coding-like issues without mentioning them. See https://github.com/httpwg/http-extensions/issues/1208#issuecomment-634351643

File: draft-ietf-httpbis-digest-headers.md (hunk 2 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:284]: The "considerations on new fields" in Semantics, asks to explicit whether intermediaries can  alter the field and how.

File: draft-ietf-httpbis-digest-headers.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:120]: Yeah its fine to keep then

File: draft-ietf-httpbis-digest-headers.md (hunk 2 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:284]:
```suggestion
```
No normative language. Discussed with @martinthomson and @mnot at interim 2020-10

**Review by LPardue — APPROVED** [review]

**Pull request merged**